### PR TITLE
[#9548] CLU and CICD for BotFramework Composer 

### PR DESCRIPTION
**Pull request by JhontSouth**
#minor

## Proposed Changes
This PR allows the use of the Composer CICD Pipeline Sample with CLU service instead of LUIS.

  - Added the _Composer CICD Pipeline Sample_ for CLU service.
  - Added the _BasicAssistantCLU_ sample.
  
## Testing
The following image shows the deployed bot working as expected.
image

### CODE REVIEW

File: composer-samples/csharp_dotnetcore/pipelines/CICDPipelineSampleCLU/BasicAssistantCLU/BasicAssistantCLU/BasicAssistantCLU.csproj [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    <UserSecretsId>74e508c9-0df9-45fe-b083-d73a77324acc</UserSecretsId>
+  </PropertyGroup>
+  <ItemGroup>
+    <Content Include="**/*.blu;**/*.dialog;**/*.lg;**/*.lu;**/*.model;**/*.onnx;**/*.qna;**/*.txt" Exclude="$(BaseOutputPath)/**;$(BaseIntermediateOutputPath)/**;wwwroot/**">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+  </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.1.8" />
+    <PackageReference Include="Microsoft.Bot.Builder.AI.Luis" Version="4.18.1" />
```
Comment (ceciliaavila) [inline composer-samples/csharp_dotnetcore/pipelines/CICDPipelineSampleCLU/BasicAssistantCLU/BasicAssistantCLU/BasicAssistantCLU.csproj]: This reference should be removed.

File: composer-samples/csharp_dotnetcore/pipelines/CICDPipelineSampleCLU/BasicAssistantCLU/BasicAssistantCLU/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            app.UseDefaultFiles();
+
+            // Set up custom content types - associating file extension to MIME type.
+            var provider = new FileExtensionContentTypeProvider();
+            provider.Mappings[".lu"] = "application/vnd.microsoft.lu";
```
Comment (ceciliaavila) [inline composer-samples/csharp_dotnetcore/pipelines/CICDPipelineSampleCLU/BasicAssistantCLU/BasicAssistantCLU/Startup.cs]: Is this necessary?

File: composer-samples/csharp_dotnetcore/pipelines/CICDPipelineSampleCLU/BasicAssistantCLU/BasicAssistantCLU/dialogs/imported/Microsoft.Bot.Components.HelpAndCancel/CancelDialog/language-understanding/en-us/CancelDialog.en-us.lu
Comment (ceciliaavila) [inline composer-samples/csharp_dotnetcore/pipelines/CICDPipelineSampleCLU/BasicAssistantCLU/BasicAssistantCLU/dialogs/imported/Microsoft.Bot.Components.HelpAndCancel/CancelDialog/language-understanding/en-us/CancelDialog.en-us.lu:1]: Shouldn't we remove these .lu files?

File: composer-samples/csharp_dotnetcore/pipelines/CICDPipelineSampleCLU/BasicAssistantCLU/BasicAssistantCLU/scripts/DeploymentTemplates/function-template-with-preexisting-rg.json [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+      "type": "bool",
+      "defaultValue": true
+    },
+    "shouldCreateAuthoringResource": {
+      "type": "bool",
+      "defaultValue": true
+    },
+    "shouldCreateLuisResource": {
+      "type": "bool",
+      "defaultValue": true
```
Comment (ceciliaavila) [inline composer-samples/csharp_dotnetcore/pipelines/CICDPipelineSampleCLU/BasicAssistantCLU/BasicAssistantCLU/scripts/DeploymentTemplates/function-template-with-preexisting-rg.json]: We should remove all LUIS references from the templates.

File: composer-samples/csharp_dotnetcore/pipelines/CICDPipelineSampleCLU/BasicAssistantCLU/BasicAssistantCLU/scripts/DeploymentTemplates/function-template-with-preexisting-rg.json [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+    "botSku": {
+      "defaultValue": "F0",
+      "type": "string",
+      "metadata": {
+        "description": "The pricing tier of the Bot Service Registration. Acceptable values are F0 and S1."
+      }
+    },
+    "luisAuthoringKey": {
+      "type": "string",
+      "defaultValue": ""
```
Comment (ceciliaavila) [inline composer-samples/csharp_dotnetcore/pipelines/CICDPipelineSampleCLU/BasicAssistantCLU/BasicAssistantCLU/scripts/DeploymentTemplates/function-template-with-preexisting-rg.json]: Same with this.

File: composer-samples/csharp_dotnetcore/pipelines/CICDPipelineSampleCLU/BasicAssistantCLU/BasicAssistantCLU/scripts/README.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+   | Resource                                      | Required/Optional |
+   | --------------------------------------------- | ----------------- |
+   | App Service plan                              | Required          |
+   | App Service                                   | Required          |
+   | Application Registration                      | Required          |
+   | Azure Cosmos DB                               | Optional          |
+   | Application Insights                          | Optional          |
+   | Azure Blob Storage                            | Optional          |
+   | LUIS authoring resource (Cognitive Services)  | Optional          |
+   | LUIS prediction resource (Cognitive Services) | Optional          |
```
Comment (ceciliaavila) [inline composer-samples/csharp_dotnetcore/pipelines/CICDPipelineSampleCLU/BasicAssistantCLU/BasicAssistantCLU/scripts/README.md]: These should be removed.

File: composer-samples/csharp_dotnetcore/pipelines/CICDPipelineSampleCLU/BasicAssistantCLU/BasicAssistantCLU/scripts/provisionComposer.js
Comment (ceciliaavila) [inline composer-samples/csharp_dotnetcore/pipelines/CICDPipelineSampleCLU/BasicAssistantCLU/BasicAssistantCLU/scripts/provisionComposer.js:1]: Is this file still used in the new pipeline? maybe we can remove it. Or remove the LUIS steps.

File: composer-samples/csharp_dotnetcore/pipelines/CICDPipelineSampleCLU/BasicAssistantCLU/BasicAssistantCLU/settings/appsettings.json [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    "enableRegexEntities": true
+  },
+  "luis": {
+    "authoringEndpoint": "",
+    "authoringRegion": "",
+    "defaultLanguage": "en-us",
+    "endpoint": "",
+    "environment": "composer",
+    "name": "BasicAssistantCLU"
+  },
```
Comment (ceciliaavila) [inline composer-samples/csharp_dotnetcore/pipelines/CICDPipelineSampleCLU/BasicAssistantCLU/BasicAssistantCLU/settings/appsettings.json]: I think we should remove these settings.

File: composer-samples/csharp_dotnetcore/pipelines/CICDPipelineSampleLUIS/BasicAssistant/BasicAssistant/dialogs/imported/Microsoft.Bot.Components.HelpAndCancel/CancelDialog/knowledge-base/en-us/CancelDialog.en-us.qna [hunk 7]
```diff
@@ -0,0 +1,3 @@
+
+
+> Imported: 68b329da9893e34099c7d8ad5cb9c940
```
Comment (ceciliaavila) [inline composer-samples/csharp_dotnetcore/pipelines/CICDPipelineSampleLUIS/BasicAssistant/BasicAssistant/dialogs/imported/Microsoft.Bot.Components.HelpAndCancel/CancelDialog/knowledge-base/en-us/CancelDialog.en-us.qna:3]: Don't remove the last blank line in files, it's a convention.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

**Review by erquirogasw — COMMENTED** [review]
Are the templates found in the scripts folder inside BassicAssistantCLU necessary? Since the original sample didn't have them.

**Review by ceciliaavila — APPROVED** [review]

**Review by sw-joelmut — APPROVED** [review]

**Pull request closed without merging**